absl/strings/escaping_test.cc

```diff
@@ -278,6 +278,98 @@ TEST_F(CUnescapeTest, UnescapesMultipleUnicodeNulls) {
                    "\0", 5), result_string_);
 }
 
+TEST(CUnescape, NonNullTerminatedOctal1) {
+  absl::string_view reference("\\111", 2);
+
+  std::string unescaped_str;
+  EXPECT_TRUE(absl::CUnescape(reference, &unescaped_str));
+
+  EXPECT_EQ(unescaped_str, "\1");
+}
+
+TEST(CUnescape, NonNullTerminatedOctal2) {
+  absl::string_view reference("\\1z1", 3);
+
+  std::string unescaped_str;
+  EXPECT_TRUE(absl::CUnescape(reference, &unescaped_str));
+
+  EXPECT_EQ(unescaped_str, "\1z");
+}
+
+TEST(CUnescape, NonNullTerminatedHex) {
```

Comment:
What do you think about naming this test `HexUnescapingDoesNotOverrunBuffer`?  Naming is hard and subjective, but that name describes what the test is trying to do a little better to me.